.travis.yml

```diff
@@ -1,31 +1,42 @@
 language: php
 
-php:
-  - 5.4
-  - 5.5
-  - 5.6
-  - 7.0
-  - 7.1
-
-env:
-  - PREFER_LOWEST="--prefer-lowest"
-  - PREFER_LOWEST=""
+sudo: false
 
 matrix:
   include:
-    # PHP 5.3 builds have to be included manually as they require precise
     - php: 5.3
       dist: precise
-      env: PREFER_LOWEST="--prefer-lowest"
+      env: WITH_LOWEST=true
     - php: 5.3
       dist: precise
-      env: PREFER_LOWEST=""
+      env: WITH_HIGHEST=true
+    - php: 5.4
+      env: WITH_LOWEST=true
+    - php: 5.4
+      env: WITH_HIGHEST=true
+    - php: 5.5
+      env: WITH_LOWEST=true
+    - php: 5.5
+      env: WITH_HIGHEST=true
+    - php: 5.6
+      env: WITH_LOWEST=true
+    - php: 5.6
+      env: WITH_HIGHEST=true
+    - php: 7.0
+      env: WITH_LOWEST=true
+    - php: 7.0
+      env: WITH_HIGHEST=true
+    - php: 7.1
+      env: WITH_LOWEST=true
+    - php: 7.1
+      env: WITH_HIGHEST=true
 
-sudo: false
+install:
+  - if [[ "$WITH_LOWEST" == "true" ]]; then composer update --prefer-lowest; fi
```

Comment:
We could also think about checking in `composer.lock` and adding builds for `WITH_LOCKED=true`:

```
- if [[ "$WITH_LOCKED" == "true" ]]; then composer install; fi
```